### PR TITLE
Move IO write buffer allocation into function.

### PR DESCRIPTION
**Pull request by ioquatix**
I was reading through the `write_lock` code and thought this would be a useful simplification.

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -1865,78 +1871,79 @@ io_binwritev(struct iovec *iov, int iovcnt, rb_io_t *fptr)
...
 
-    if (fptr->wbuf.ptr == NULL && !(fptr->mode & FMODE_SYNC)) {
-	fptr->wbuf.off = 0;
-	fptr->wbuf.len = 0;
-	fptr->wbuf.capa = IO_WBUF_CAPA_MIN;
-	fptr->wbuf.ptr = ALLOC_N(char, fptr->wbuf.capa);
-	fptr->write_lock = rb_mutex_new();
-	rb_mutex_allow_trap(fptr->write_lock, 1);
-    }
+    io_allocate_write_buffer(fptr, 0);
```
Comment (nobu) [inline io.c:1874]:
```suggestion
    io_allocate_write_buffer(fptr, 1);
```

**Comment by nobu** [comment]
The rest of the first three hunks seem irrelevant and unnecessary.

**Comment by ioquatix** [comment]
@nobu I fixed the whitespace while reading the code.

**Comment by nobu** [comment]
It feels just confusing.

**Comment by ioquatix** [comment]
What is confusing?

**Pull request merged**